### PR TITLE
fix degree bound checks

### PR DESCRIPTION
**Pull request by akinovak**

### CODE REVIEW

File: src/prover.rs [hunk 1]
```diff
@@ -445,7 +446,7 @@ mod prover_rounds_tests {
         // check b0 degree
         {
             let lhs_0 = b0_cm;
-            let rhs_0 = srs_g2[table.size - witness.size - 1];
+            let rhs_0 = srs_g2[table.size - witness.size + 1];
```
Comment (kobigurk) [inline src/prover.rs:449]:
```suggestion
            let rhs_0 = srs_g2[state.table.size - 1 - (state.witness.size - 2)];
```

File: src/verifier.rs [hunk 2]
```diff
@@ -24,7 +24,7 @@ impl<E: PairingEngine> VerifierKey<E> {
     pub fn new(srs_g2: &[E::G2Affine], table_size: usize, witness_size: usize) -> Self {
         Self {
             x: srs_g2[1].into(),
-            x_pow_b0_bound: srs_g2[table_size - witness_size - 1].into(),
+            x_pow_b0_bound: srs_g2[table_size - witness_size + 1].into(),
```
Comment (kobigurk) [inline src/verifier.rs:27]:
```suggestion
            x_pow_b0_bound: srs_g2[state.table.size - 1 - (state.witness.size - 2)].into(),
```

**Comment by kobigurk** [comment]
@akinovak let's open a followup issue to generalize this to SRS size rather than table size?

**Review by kobigurk — CHANGES_REQUESTED** [review]

**Review by kobigurk — APPROVED** [review]

**Comment by akinovak** [comment]
> @akinovak let's open a followup issue to generalize this to SRS size rather than table size?

I'll leave this to be identical as in paper. We can keep maybe separate branch with that feature, wdyt?

**Comment by kobigurk** [comment]
> > @akinovak let's open a followup issue to generalize this to SRS size rather than table size?
> 
> I'll leave this to be identical as in paper. We can keep maybe separate branch with that feature, wdyt?

A separate branch will add clutter and quickly go out of date imo.
good point to leave the same as paper, at least for now.

maybe later we could find a way to generalize, if anyone needs it.

**Pull request merged**